scripts/test/launchTestUtilities.ts

```diff
@@ -17,8 +18,121 @@ const minimum = 'minimum'
+ * setup vscode and then run tests. An important thing to note is that in CI we do not have VS Code installed,
+ * so this script needs to do this itself.
+ *
+ * If you want to run the tests manually you should use the `Run & Debug` menu in VS Code instead.
+ */
+export async function runToolkitTests(suite: SuiteName, relativeTestEntryPoint: string) {
+    try {
+        console.log(`Running ${suite} test suite...`)
+
+        const args = await getVSCodeCliArgs({
+            vsCodeExecutablePath: await setupVSCodeInstance(suite),
+            relativeTestEntryPoint,
+            suite,
+        })
+        console.log(`runTests() args:\n${JSON.stringify(args, undefined, 2)}`)
+        const result = await runTests(args)
+
+        console.log(`Finished running ${suite} test suite with result code: ${result}`)
+        process.exit(result)
+    } catch (err) {
+        console.error(err)
+        console.error('Failed to run tests')
+        process.exit(1)
+    }
+}
+
+/**
+ * Resolves all args for {@link runTests}
+ */
+async function getVSCodeCliArgs(params: {
+    vsCodeExecutablePath: string
+    relativeTestEntryPoint: string
+    suite: SuiteName
+}): Promise<TestOptions> {
+    const projectRootDir = process.cwd()
+
+    let disableExtensionsArgs: string[] = []
+    let disableWorkspaceTrustArg: string[] = []
+
+    if (params.suite !== 'unit') {
+        disableExtensionsArgs = await getCliArgsToDisableExtensions(params.vsCodeExecutablePath, {
+            except: [
+                VSCODE_EXTENSION_ID.python,
+                VSCODE_EXTENSION_ID.yaml,
+                VSCODE_EXTENSION_ID.jupyter,
+                VSCODE_EXTENSION_ID.go,
+                VSCODE_EXTENSION_ID.java,
+                VSCODE_EXTENSION_ID.javadebug,
+                VSCODE_EXTENSION_ID.git,
+                VSCODE_EXTENSION_ID.remotessh,
+            ],
+        })
+        disableWorkspaceTrustArg = [disableWorkspaceTrust]
+    }
+
+    const workspacePath = join(projectRootDir, 'dist', 'src', 'testFixtures', 'workspaceFolder')
+
+    return {
+        vscodeExecutablePath: params.vsCodeExecutablePath,
+        extensionDevelopmentPath: projectRootDir,
+        extensionTestsPath: resolve(projectRootDir, params.relativeTestEntryPoint),
+        launchArgs: [...disableExtensionsArgs, workspacePath, ...disableWorkspaceTrustArg],
+        extensionTestsEnv: {
+            ['DEVELOPMENT_PATH']: projectRootDir,
+            ['AWS_TOOLKIT_AUTOMATION']: params.suite,
+        },
+    }
+}
+
+/**
+ * Alternative to the `--disable-extensions` CLI flag which allows us to
+ * selectively keep some extensions enabled.
+ *
+ * @param vsCodeExecutablePath Path to vscode CLI program.
+ * @param exceptIds List of extension ids that should *not* be disabled.
+ *
+ * @returns List of args which the caller is expected to pass to vscode CLI, of the form:
+ * ["--disable-extension", "foo.bar.baz", "--disable-extension", ...]
+ */
+async function getCliArgsToDisableExtensions(
```

Comment:
this and `setupVSCodeInstance` look almost the same as the old code. is there any way to avoid the diff-churn?

Reply:
fixed